### ISSUE TITLE
autohook_fifo_**** directories not cleaned up when using VSCode git integration

### ISSUE THREAD

**Issue by sciyoshi** [p7g/Autohook#7]
## What's the issue?
Using `git checkout` commands in VSCode cause these directories not to be cleaned up, and a bunch of background processes don't get cleaned up:

```
sciyoshi 31015  0.0  0.0  14464  3848 tty2     S+   00:00   0:00 git checkout -q -b dhall --no-track HEAD
sciyoshi 31016  0.0  0.0  10212  4020 tty2     S+   00:00   0:00  \_ bash .git/hooks/post-checkout 7f32868953a15638f89da919d535d2c438c4541c 7f32868953a15638f89da919d535d2c438c4541c 1
sciyoshi 31031  0.0  0.0  10212  2564 tty2     S+   00:00   0:00      \_ bash .git/hooks/post-checkout 7f32868953a15638f89da919d535d2c438c4541c 7f32868953a15638f89da919d535d2c438c4541c 1
sciyoshi  6506  0.0  0.0 312312  8376 tty2     S+   00:05   0:00 git checkout -q
sciyoshi  6515  0.0  0.0  10212  4004 tty2     S+   00:05   0:00  \_ bash .git/hooks/post-checkout dc38beb12103d57e8b9cb9bc53b5fcf735a18433 dc38beb12103d57e8b9cb9bc53b5fcf735a18433 1
sciyoshi  6530  0.0  0.0  10212  2552 tty2     S+   00:05   0:00      \_ bash .git/hooks/post-checkout dc38beb12103d57e8b9cb9bc53b5fcf735a18433 dc38beb12103d57e8b9cb9bc53b5fcf735a18433 1
sciyoshi  6561  0.0  0.0 115704  8284 tty2     S+   00:05   0:00 git checkout -q
sciyoshi  6570  0.0  0.0  10212  3804 tty2     S+   00:05   0:00  \_ bash .git/hooks/post-checkout dc38beb12103d57e8b9cb9bc53b5fcf735a18433 dc38beb12103d57e8b9cb9bc53b5fcf735a18433 1
sciyoshi  6585  0.0  0.0  10212  2496 tty2     S+   00:05   0:00      \_ bash .git/hooks/post-checkout dc38beb12103d57e8b9cb9bc53b5fcf735a18433 dc38beb12103d57e8b9cb9bc53b5fcf735a18433 1
sciyoshi  6613  0.0  0.0 312312  8244 tty2     S+   00:05   0:00 git checkout -q
sciyoshi  6622  0.0  0.0  10212  3848 tty2     S+   00:05   0:00  \_ bash .git/hooks/post-checkout dc38beb12103d57e8b9cb9bc53b5fcf735a18433 dc38beb12103d57e8b9cb9bc53b5fcf735a18433 1
sciyoshi  6637  0.0  0.0  10212  2600 tty2     S+   00:05   0:00      \_ bash .git/hooks/post-checkout dc38beb12103d57e8b9cb9bc53b5fcf735a18433 dc38beb12103d57e8b9cb9bc53b5fcf735a18433 1
sciyoshi  6658  0.0  0.0 312312  8292 tty2     S+   00:05   0:00 git checkout -q
sciyoshi  6669  0.0  0.0  10212  4096 tty2     S+   00:05   0:00  \_ bash .git/hooks/post-checkout dc38beb12103d57e8b9cb9bc53b5fcf735a18433 dc38beb12103d57e8b9cb9bc53b5fcf735a18433 1
sciyoshi  6684  0.0  0.0  10212  2608 tty2     S+   00:05   0:00      \_ bash .git/hooks/post-checkout dc38beb12103d57e8b9cb9bc53b5fcf735a18433 dc38beb12103d57e8b9cb9bc53b5fcf735a18433 1
sciyoshi 30193  0.0  0.0   8860   924 pts/3    S+   09:16   0:00      \_ grep --color=auto git
```

**Comment by p7g** [comment]
Was able to reproduce this, and fixed it in [`43c278c`](https://github.com/p7g/Autohook/commit/43c278c36608c3f94d9c414544c1ab866cfc8fa4). The cause was some sloppy clean up which didn't happen if there were no scripts to run (for example if there were some staged files without any scripts defined for their extension).

**Comment by p7g** [comment]
This still seems to happen sometimes, so I'm gonna reopen this and take another look

**Comment by p7g** [comment]
So this should be fixed for real by [df6388a](https://github.com/p7g/Autohook/commit/df6388a711b9dbbf62fce20664e6fad204384690), but I'll leave this open for a while in case there are some more cases where this happens.
